### ISSUE TITLE
Errors should consistently include filename, line number, and column number

### ISSUE THREAD

**Issue by casey** [casey/just#1277]
I'd like to copy Rust's error messages, and have a line after the error message that starts with a ` -->` and includes the filename, line number, and column number. So this error:

```
error: Recipe `foo` failed on line 2 with exit code 1
```

From this justfile:

```
foo:
  exit 1
```

Should look like this:

```
error: Recipe `foo` failed on line 2 with exit code 1
 --> justfile:2:3
```

This would allow tools to easily extract the error information and jump to the right place in the justfile.

Also, this would make the transition to any kind of multi-justfile world (e.g. includes, modules, etc) much easier, since we would already have the functionality for printing out the filename.

**Comment by BeatScherrer** [comment]
I try to work up to this since that might already be useful for me.

**Comment by casey** [comment]
@beatscherrer Nice! I'd start with tests, add those, and then work from there. You can even open a draft PR with the failing tests, to confirm that they look good.

**Comment by Eloitor** [comment]
What about allowing a message in case of failure? Something like python's assert 
```
assert expression[, assertion_message]
```

I don't know what sintax to suggest, but maybe something like:
```
foo:
  exit 1
  > echo "X failed, do Y and try again"
```

**Comment by casey** [comment]
Fixed in #1737.

**Comment by BeatScherrer** [comment]
> Fixed in #1737.

That's really cool to see. I tried to work on this but got lost on how to implement it in the best way since I'm neither familiar with the just code base nor a proficient rust developer.
